estuary_updater/handlers/koji.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from __future__ import unicode_literals, absolute_import
+
+from estuary.models.koji import KojiBuild, KojiTag
+from estuary.models.distgit import DistGitCommit
+
+from estuary_updater.handlers.base import BaseHandler
+
+
+class KojiHandler(BaseHandler):
+    """A handler for Koji related messages."""
+
+    @staticmethod
+    def can_handle(msg):
+        """
+        Determine if this handler can handle this message.
+
+        :param dict msg: a message to be analyzed
+        :return: a bool based on if the handler can handle this kind of message
+        :rtype: bool
+        """
+        supported_topics = [
+            '/topic/VirtualTopic.eng.brew.build.complete',
+            '/topic/VirtualTopic.eng.brew.build.tag',
+            '/topic/VirtualTopic.eng.brew.build.untag'
+        ]
+        return msg['topic'] in supported_topics
+
+    def handle(self, msg):
+        """
+        Handle a message and update Neo4j if necessary.
+
+        :param dict msg: a message to be processed
+        """
+        topic = msg['topic']
+
+        if topic == '/topic/VirtualTopic.eng.brew.build.complete':
+            self.build_complete_handler(msg)
+        elif topic == '/topic/VirtualTopic.eng.brew.build.tag' or \
+                topic == '/topic/VirtualTopic.eng.brew.build.untag':
+            self.build_tag_handler(msg)
+        else:
+            raise RuntimeError('This message is unable to be handled: {0}'.format(msg))
+
+    def build_complete_handler(self, msg):
+        """
+        Handle a build complete message and update Neo4j if necessary.
+
+        :param dict msg: a message to be processed
+        """
+        build = self.get_or_create_build(msg['body']['msg']['info']['id'])
+
+        commit_hash = msg['body']['msg']['info']['source'].split('#')[1]
```

Comment:
This will work in cases where the build is an RPM or container image, but this won't work for some other types. I'll send you a message showing this.

Reply:
My suggestion would be to ignore non-RPM and container image builds like we do in the scraper but we'll have to find a way to do that (likely going to have to look at the build task).